.github/workflows/pythonpublish.yml

```diff
@@ -13,24 +13,34 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
+
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
           python-version: "3.x"
 
+      - name: Verify version
+        shell: python
+        run: |
+          import tomllib
+          expected = "${{ github.event.release.tag_name }}"
+          with open("pyproject.toml", "rb") as fp:
+              version = tomllib.load(fp)["project"]["version"]
+              if version != expected:
+                  exit(f"Defined version {version} does not match expected version {expected}")
+
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install setuptools wheel twine
+          pip install setuptools build
 
-      - name: Verify version
+      - name: Build
```

Comment:
This was split out from publish as there is no need for the build step to have access to the password secret.